_transcripts/032-rust-1.48-1.49.md

```diff
@@ -0,0 +1,731 @@
+anyone thinks of a reason why future pending is useful outside of just some very
+trivial tests, then please let us know and we'll try to make sure to talk about
+this in the next episode as to why I was wrong and why `future::pending` is
+essential.
+
+__Ben__: The only remaining thing with the APIs in this release is that some
+APIs have been made `const`, and so this is just not const generics, just const
+evaluation.
+
+__Jon__: What? That's entirely new.
+
+__Ben__: This happens every single release. There's always a few things that
+become `const`.
+
+__Jon__: What? We have `const fn`s?
+
+__Ben__: No need to be so incredulous. This time, they are a few `Option`
+methods, so `Option::is_some`, `Option::is_none`. And so if you have an `Option`
+you can use `if my_option.is_some()` to test if it's `Some` or `None`, and this
+is a bit redundant with the `if let` syntax, if you're familiar with that. I
+believe these methods predate `if let`. They are ancient. They've been around
+since `Option` has existed. But now it's cool that they are `const`, because if
+you wanted to use— I believe, was it last release, or the previous one, where
+now you can use branches, `if`s in `const fn`s, And so now these are kind of
+just expanding on those building blocks. Now the things you might want to use
+with those `if` statements are now becoming `const`. So we're gradually seeing
+more and more of the building blocks of all Rust code becoming `const`. Which
+opens up more and more chances for more things to become `const`. So ongoing
+const- ification of `std` continues unabated.
+
+__Jon__: I think it might even be that it's because the const-ification work
+happened to enable conditionals. That's how these functions are able to be
+`const` now. Right? Because if you think about internally, `is_some` and
+`is_none` sort of need to branch on whether the inner type is `Some` or `None`,
+right?
+
+__Ben__: I imagine they're a `match` statement internally.
+
+__Jon__: Yeah, exactly.
+
+__Ben__: Same with `Result::is_ok`, `Result::is_err`, and then a few other
+things: `Ordering::reverse`, `Ordering::then`. Did you have something else to
+say?
+
+__Jon__: Yeah, I was just going to say, I think what we're seeing with most of
+these is, as you mentioned, like, the continuing trend from before, I do wonder,
+like, what the next step is for const-ification. We got some limited amount of
+looping and conditionals. Are we going to get, like, compile time heap memory or
+something, soon? Like, that would be really cool. Like, what would it take to
+make, like—
+
+__Ben__: Maybe we should interview someone who's working on this and find out?
+
+__Jon__: Oh, yeah. We should chat to Oli for example. That'd be really cool. Or
+if you, as the listener, want to talk to Oli about this, you should reach out,
+and we will happily help you make an episode with Oli where we talk about const-
+ification. That sounds great.
```

Comment:
Just want to confirm that "Oli" is the right name?

Reply:
Indeed. This is @oli-obk